xcm/xcm-simulator/example/src/parachain.rs

```diff
@@ -266,7 +272,8 @@ pub mod mock_msg_queue {
 						Self::deposit_event(Event::UnsupportedVersion(id));
 					},
 					Ok(Ok(x)) => {
-						let outcome = T::XcmExecutor::execute_xcm(Parent.into(), x, limit);
+						let outcome = T::XcmExecutor::execute_xcm(Parent.into(), x.clone(), limit);
+						<ReceivedDmp<T>>::mutate(|y| y.push((Parent.into(), x)));
```

Comment:
```suggestion
						<ReceivedDmp<T>>::append((Parent.into(), x));
```

But with the `MultiLocation` changes above, you might need this instead:

```suggestion
						<ReceivedDmp<T>>::append(x);
```